pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3274,3 +3244,70 @@ fn dirty_storage_with_tx() {
             );
         });
 }
+
+#[test]
+fn reject_failed_instruction() {
+    ExtBuilder::default().build().execute_with(|| {
+        let alice = User::new(AccountKeyring::Alice);
+        let bob = User::new(AccountKeyring::Bob);
+
+        let token_name = b"ACME";
+        let ticker = Ticker::try_from(&token_name[..]).unwrap();
+        let venue_counter = init(token_name, ticker, AccountKeyring::Alice.public());
+        let amount = 100u128;
+
+        let instruction_counter = create_instruction(&alice, &bob, venue_counter, ticker, amount);
+
+        assert_ok!(Settlement::affirm_instruction(
+            bob.origin(),
+            instruction_counter,
+            default_portfolio_vec(bob.did),
+            1
+        ));
+
+        next_block();
+        assert_eq!(
+            Settlement::instruction_details(instruction_counter).status,
+            InstructionStatus::Failed,
+        );
```

Comment:
It would be good to write down why it is that this resulted in a failed status. I'm having a hard time following along without more deeply understanding the execution logic.

General principle around tests (or code): try to reduce the amount of time a reader has to spend finding the reason as to why something works, or what the code is trying to do, and instead allow the reader to simply verify that the method the comment outlines does reach the outcome. That is, we want to go from a mental decision problem to a verification problem, where the latter is easier, for computers and humans alike.